### PR TITLE
SpanishGameClass

### PR DESCRIPTION
**Pull request by danielgwj**
Note: I have comment most of the English parts out

Also, I have made the Game class into an abstract class

### CODE REVIEW

**Comment by danielgwj** [comment]
@matsuish I would encourage you to take note at some of the changes that I have made.
The EnglishGame class can be made based on the SpanishGame class.

Also a few other files to note are:
- ApplicationController
- Routes

Also do comment on my code and ask questions if you feel a bit lost.
I know there are lots of code to change so it can be quite confusing at first.

**Comment by danielgwj** [comment]
@jrotithor By changing the Game() class to an abstract class, the child classes such as EnglishGame() and SpanishGame() will be able to inherit the functions within the abstract class

File: src/main/java/models/Game.java [hunk 1]
```diff
@@ -7,25 +7,27 @@
...
 
-    public int cardsLeft;
+    public int cardsLeft = 0;
 
     public Game(){
         cols.add(new ArrayList<Card>());
         cols.add(new ArrayList<Card>());
         cols.add(new ArrayList<Card>());
         cols.add(new ArrayList<Card>());
+        this.buildDeck();
```
Comment (matsuish) [inline src/main/java/models/Game.java:25]: the reason why shuffle() is not here is because of unit tests?

Comment (danielgwj) [inline src/main/java/models/Game.java:25]: the shuffle() is just below the buildDecl method

Also the shuffle method is called after initializing the game in the Application Controller

Comment (matsuish) [inline src/main/java/models/Game.java:25]: Yeah, I am wondering why shuffle() is not called in constructor but buildDeck() is.
originally both were called in ApplicationController.

Comment (danielgwj) [inline src/main/java/models/Game.java:25]: shuffle() wasnt called in the constructor before. It has always been called in the gameGet() method in the Application controller

right now, i have placed the buildDeck() in the Game class constructor because I've made the Game class into an abstract class, and initialized buildDeck() as an abstract method.

This abstract method will be different in differrent child cases and will be called when the Game() constructor is called.

Basically i've placed it there because I didn't want to create another buildDeckSpanish() method. And since we have decided to use the Template inheritence design pattern, this would be the way to go to show that we are using inheritence in our project! :D

Sorry if this posts confuses you as it is quite long!
Shoot me more questions if your still in doubt!

Comment (matsuish) [inline src/main/java/models/Game.java:25]: Thank you for your reply. And sorry for making confusing you. My English must be terrible.

I am sorry if I am wrong, but you may not notice that my comment is line note (comment in "Files changed" page).
I was just curious that you didn't put shuffle() in the constructor, while you put buildDeck() in it.
I think there will be no difference if put them(shuffle and buildDeck) in constructor or call them in application controller, so I wondered why.
